diff --git a/node_modules/@react-native-documents/picker/ios/swift/FileOperations.swift b/node_modules/@react-native-documents/picker/ios/swift/FileOperations.swift
index 6f1682f..9ddf3d6 100644
--- a/node_modules/@react-native-documents/picker/ios/swift/FileOperations.swift
+++ b/node_modules/@react-native-documents/picker/ios/swift/FileOperations.swift
@@ -13,8 +13,8 @@ import Foundation
   
   static func moveFiles(from: [[String: String]], destinationPreset: String) -> [[String: String?]] {
     let destinationRootDir = getDirectoryForFileDestination(destinationPreset)
-    let uniqueSubDirName = UUID().uuidString
-    let destinationDir = destinationRootDir.appendingPathComponent(uniqueSubDirName, isDirectory: true)
+//    let uniqueSubDirName = UUID().uuidString
+    let destinationDir = destinationRootDir.appendingPathComponent("Ringtones", isDirectory: true)
     
     do {
       try FileManager.default.createDirectory(at: destinationDir, withIntermediateDirectories: true, attributes: nil)
