diff --git a/node_modules/react-native-webrtc/android/src/main/java/com/oney/WebRTCModule/WebRTCView.java b/node_modules/react-native-webrtc/android/src/main/java/com/oney/WebRTCModule/WebRTCView.java
index 124701b..30dfbfd 100644
--- a/node_modules/react-native-webrtc/android/src/main/java/com/oney/WebRTCModule/WebRTCView.java
+++ b/node_modules/react-native-webrtc/android/src/main/java/com/oney/WebRTCModule/WebRTCView.java
@@ -421,7 +421,7 @@ public class WebRTCView extends ViewGroup {
      * @param streamURL The URL of the {@code MediaStream} to be rendered by
      * this {@code WebRTCView} or {@code null}.
      */
-    void setStreamURL(String streamURL) {
+    public void setStreamURL(String streamURL) {
         // Is the value of this.streamURL really changing?
         if (!Objects.equals(streamURL, this.streamURL)) {
             // XXX The value of this.streamURL is really changing. Before
