diff --git a/android/src/main/java/com/oney/WebRTCModule/WebRTCView.java b/android/src/main/java/com/oney/WebRTCModule/WebRTCView.java
index 5d14d0f76460ee9a73086fca706e3cf70e8ab631..d2c08ab3cae57380aee9b377895da1f01f36fc0f 100644
--- a/android/src/main/java/com/oney/WebRTCModule/WebRTCView.java
+++ b/android/src/main/java/com/oney/WebRTCModule/WebRTCView.java
@@ -451,7 +451,7 @@ public class WebRTCView extends ViewGroup {
      * @param streamURL The URL of the {@code MediaStream} to be rendered by
      * this {@code WebRTCView} or {@code null}.
      */
-    void setStreamURL(String streamURL) {
+    public void setStreamURL(String streamURL) {
         // Is the value of this.streamURL really changing?
         if (!Objects.equals(streamURL, this.streamURL)) {
             // XXX The value of this.streamURL is really changing. Before
